### PR TITLE
Changes since last PR: create a common ancestor to LogicalCollection/LogicalView

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

**Comment by danielhlarkin** [comment]
@vasiliy-arangodb Could you give me a little context for why you are creating this `LogicalDataSource` class?

File: arangod/VocBase/LogicalCollection.cpp [hunk 1]
```diff
@@ -1381,37 +1393,41 @@ Result LogicalCollection::compareChecksums(VPackSlice checksumSlice, std::string
...
-      result.append(std::to_string(_cid));
+      result.append(std::to_string(id()));
     }
   }
   return result;
 }
+
+// -----------------------------------------------------------------------------
+// --SECTION--                                                       END-OF-FILE
+// -----------------------------------------------------------------------------
```
Comment (danielhlarkin) [inline arangod/VocBase/LogicalCollection.cpp]: Is there a particular reason you are adding these footers everywhere?

File: arangod/VocBase/LogicalDataSource.h [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+  inline bool deleted() const noexcept { return _deleted; }
+  virtual void drop() = 0;
+  inline TRI_voc_cid_t id() const { return _id; }
+  inline std::string const& name() const { return _name; }
+  inline TRI_voc_cid_t planId() const noexcept { return _planId; }
+  virtual Result rename(std::string&& newName, bool doSync) = 0;
+  inline Type const& type() const noexcept { return _type; }
+  inline TRI_vocbase_t* vocbase() const { return _vocbase; }
+
```
Comment (danielhlarkin) [inline arangod/VocBase/LogicalDataSource.h]: The use of `noexcept` here seems rather inconsistent. For instance, why is `planId()` `noexcept`, but `id()` is not? We haven't come to a consensus as a team yet that we want to use `noexcept` on a regular basis, so I'd lean toward removing the specifiers here, at least for the time being.

File: arangod/VocBase/LogicalCollection.cpp (hunk 1 above)
Comment (vasiliy-arangodb) [reply arangod/VocBase/LogicalCollection.cpp]: I find them convenient when scrolling through long files or examining diffs, to know if I missed anything at the end of the file. They do not serve any other purpose.

File: arangod/VocBase/LogicalDataSource.h (hunk 2 above)
Comment (vasiliy-arangodb) [reply arangod/VocBase/LogicalDataSource.h]: I'll address the noexcept consistency issue in one of the subsequent pull-requests for this task, there was divergence purely due to copy-paste  reasons, thank you for catching it

**Pull request merged**